misc/diagram-util/src/main/java/org/ballerinalang/diagramutil/DiagramUtil.java

```diff
@@ -53,6 +55,23 @@ public static JsonElement getSyntaxTreeJSON(Document srcFile, SemanticModel sema
         return syntaxTreeJson;
     }
 
+    public static JsonElement getSyntaxTreeJSONByRange(Node node, SemanticModel semanticModel) {
+        JsonElement syntaxTreeJson;
+        try {
+            SyntaxTreeMapGenerator mapGenerator = new SyntaxTreeMapGenerator(semanticModel);
+
+            if (node.kind() == SyntaxKind.LIST) {
+                syntaxTreeJson = mapGenerator.transformSyntaxNode(node.parent());
+            } else {
+                syntaxTreeJson = mapGenerator.transformSyntaxNode(node);
+            }
```

Comment:
maybe we can simplify this by setting the node to be evaluated with a single liner if check and then use the assign once